numba_cuda/numba/cuda/tests/test_analysis.py

```diff
@@ -0,0 +1,1169 @@
+        # **kwargs:
+        #        - flags: args to pass to `jit` default is `nopython=True`,
+        #          e.g. permits use of e.g. object mode.
+
+        func_ir = compile_to_ir(func)
+        before = func_ir.copy()
+        if self._DEBUG:
+            print("=" * 80)
+            print("before inline")
+            func_ir.dump()
+
+        # run closure inlining to ensure that nonlocals in closures are visible
+        inline_pass = InlineClosureCallPass(
+            func_ir,
+            ParallelOptions(False),
+        )
+        inline_pass.run()
+
+        # Remove all Dels, and re-run postproc
+        post_proc = postproc.PostProcessor(func_ir)
+        post_proc.run()
+
+        rewrite_semantic_constants(func_ir, args_tys)
+        if self._DEBUG:
+            print("=" * 80)
+            print("before prune")
+            func_ir.dump()
+
+        dead_branch_prune(func_ir, args_tys)
+
+        after = func_ir
+        if self._DEBUG:
+            print("after prune")
+            func_ir.dump()
+
+        before_branches = self.find_branches(before)
+        self.assertEqual(len(before_branches), len(prune))
+
+        # what is expected to be pruned
+        expect_removed = []
+        for idx, prune in enumerate(prune):
+            branch = before_branches[idx]
+            if prune is True:
+                expect_removed.append(branch.truebr)
+            elif prune is False:
+                expect_removed.append(branch.falsebr)
+            elif prune is None:
+                pass  # nothing should be removed!
+            elif prune == "both":
+                expect_removed.append(branch.falsebr)
+                expect_removed.append(branch.truebr)
+            else:
+                assert 0, "unreachable"
+
+        # compare labels
+        original_labels = set([_ for _ in before.blocks.keys()])
+        new_labels = set([_ for _ in after.blocks.keys()])
+        # assert that the new labels are precisely the original less the
+        # expected pruned labels
+        try:
+            self.assertEqual(new_labels, original_labels - set(expect_removed))
+        except AssertionError as e:
+            print("new_labels", sorted(new_labels))
+            print("original_labels", sorted(original_labels))
+            print("expect_removed", sorted(expect_removed))
+            raise e
+
+        if [
+            arg is types.NoneType("none") or arg is types.Omitted(None)
+            for arg in args_tys
+        ].count(True) == 0:
+            self.run_func(func, args)
+
+    def run_func(self, impl, args):
+        cres = jit(impl)
+        dargs = args
+        out = np.zeros(1)
+        cout = np.zeros(1)
+        args += (out,)
+        dargs += (cout,)
+        cres.py_func(*args)
+        cres[1, 1](*dargs)
+        self.assertPreciseEqual(out[0], cout[0])
+
+
+class TestBranchPrune(TestBranchPruneBase):
+    @skip_on_cudasim("Analysis passes not done in simulator")
```

Comment:
AFAICT this whole file is skipped under the simulator so we might just use a global skip here